modules/data-loading/partials/kafka/kafka-data-source-details.adoc

```diff
@@ -13,62 +13,239 @@ To configure the data source object, the minimum requirement is the address of t
+
+  "producer.security.protocol": "SSL",
+  "producer.ssl.endpoint.identification.algorithm": "none",
+  "producer.ssl.keystore.location": "/path/to/client.keystore.jks",
+  "producer.ssl.keystore.password": "******",
+  "producer.ssl.key.password": "******",
+  "producer.ssl.truststore.location": "/path/to/client.truststore.jks",
+  "producer.ssl.truststore.password": "******"
+}
+----
+
+===== SASL_PLAINTEXT
+Need to configure:
+
+* `<prefix>.security.protocol`
+* `<prefix>.sasl.<field>`
+
+[source,json,linenum]
+----
+{
+  "type": "mirrormaker",
+  "source.cluster.bootstrap.servers": "<KAFKAURL>:<PORT>",
+
+  "consumer.security.protocol": "SASL_PLAINTEXT",
+  "consumer.sasl.mechanism": "<GSSAPI/PLAIN/etc>",
+
+  "source.admin.security.protocol": "SASL_PLAINTEXT",
+  "source.admin.sasl.mechanism": "<GSSAPI/PLAIN/etc>",
+
+  "producer.security.protocol": "SASL_PLAINTEXT",
+  "producer.sasl.mechanism": "<GSSAPI/PLAIN/etc>",
+}
+----
+
+===== SASL_SSL
+Need to configure:
+
+* `<prefix>.security.protocol`
+* `<prefix>.sasl.<field>`
+* `<prefix>.ssl.<field>`
+
+[source,json,linenum]
+----
+{
+  "type": "mirrormaker",
+  "source.cluster.bootstrap.servers": "<KAFKAURL>:<PORT>",
+
+  "consumer.security.protocol": "SASL_SSL",
+  "consumer.sasl.mechanism": "<GSSAPI/PLAIN/etc>",
+  "consumer.ssl.endpoint.identification.algorithm": "none",
+  "consumer.ssl.keystore.location": "/path/to/client.keystore.jks",
+  "consumer.ssl.keystore.password": "******",
+  "consumer.ssl.key.password": "******",
+  "consumer.ssl.truststore.location": "/path/to/client.truststore.jks",
+  "consumer.ssl.truststore.password": "******",
+
+  "source.admin.security.protocol": "SASL_PLAINTEXT",
+  "source.admin.sasl.mechanism": "<GSSAPI/PLAIN/etc>",
+  "source.admin.ssl.endpoint.identification.algorithm": "none",
+  "source.admin.ssl.keystore.location": "/path/to/client.keystore.jks",
+  "source.admin.ssl.keystore.password": "******",
+  "source.admin.ssl.key.password": "******",
+  "source.admin.ssl.truststore.location": "/path/to/client.truststore.jks",
+  "source.admin.ssl.truststore.password": "******",
+
+  "producer.security.protocol": "SASL_PLAINTEXT",
+  "producer.sasl.mechanism": "<GSSAPI/PLAIN/etc>",
+  "producer.ssl.endpoint.identification.algorithm": "none",
+  "producer.ssl.keystore.location": "/path/to/client.keystore.jks",
+  "producer.ssl.keystore.password": "******",
+  "producer.ssl.key.password": "******",
+  "producer.ssl.truststore.location": "/path/to/client.truststore.jks",
+  "producer.ssl.truststore.password": "******"
+}
+----
+
+===== Third Party SASL Mechanism
+For both `SASL` and `SASL_SSL` when a third party mechanism it is necessary to:
+
+* Include the `<prefix>.sasl.jaas.config` in addition to the `<prefix>.sasl.client.callback.handler.class` in the configuration
+* Place the third party jar under `$(gadmin config get System.Approot)/kafka/libs/`
+
+[source,json,linenum]
+.Example SASL Configuration with third party mechanism
+----
+{
```

Comment:
@pingxieTG My intention here is to give a proper example of a third party security mechanism for SASL. What optional arguments do we want to show here instead of <GSSAPI/PLAIN/etc> ?

I think the other fields are okay as generics as they are explained in detail in the table of parameters

Reply:
Pls check this example: https://github.com/aws/aws-msk-iam-auth?tab=readme-ov-file#configuring-a-kafka-client-to-use-aws-iam-with-aws_msk_iam-mechanism